### ISSUE TITLE
Improve expand/collapse rules within a project

### ISSUE THREAD

**Issue by distefam** [NorfairKing/smos#212]
When entering a project it is often difficult to get one's bearings, particularly if that project contains many items and especially when some of those items are nested.

Personally, I want to be able to see any items with a `NEXT` state irrespective of their nesting level. Currently, if I have a `NEXT` item that is underneath a parent with a `TODO` state for example I will not see that `NEXT` item without expanding its parent. 

Additionally, I would like all completed items to be collapsed so that I can focus on what is actionable.

So, to summarize:

- Expand parent when a child has a `NEXT` state (maybe also if a child is `TODO`, `WAITING`, or `READY`)
- Collapse any item with a state of `DONE`, `FAILED`, or `CANCELLED` (collectively I'll call these "completed" items)
- Hide any notes associated with any completed items

**Comment by NorfairKing** [comment]
Looks good. One more question: where should the cursor be, to start?

**Comment by distefam** [comment]
On the first (top) next item in the project. And, in the absense of a next action the top todo action.

On May 24, 2021 2:29:31 PM EDT, Tom Sydney Kerckhove ***@***.***> wrote:
>Looks good. One more question: where should the cursor be, to start?
>
>-- 
>You are receiving this because you modified the open/close state.
>Reply to this email directly or view it on GitHub:
>https://github.com/NorfairKing/smos/issues/212#issuecomment-847244663

-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.
